common/models/types/index.d.ts

```diff
@@ -114,6 +114,12 @@ declare interface LexicalModel {
    */
   predict(transform: Transform, context: Context): Distribution<Suggestion>;
 
+  /**
+   * Tokenizes the current context state, returning an array of all resulting tokens.
+   * @param context 
+   */
+  tokenize(context: Context): USVString[];
```

Comment:
I don't think that this should be part of the model's interface, since developers of custom models will need to implement this themselves. Chances are, they'll provide the exact same implementation as the one here in the trie model. Can this functionality be placed elsewhere?

Reply:
Only if we do the same for `wordbreak`.  Otherwise, we're being quite inconsistent... and I'd like to point out that given `tokenize`, we can easily get `wordbreak` for free.

Reply:
https://github.com/keymanapp/keyman/blob/1cbf845b00553389361f71b0ef76dac2f93b3a20/common/models/types/index.d.ts#L117-L132